BMSForTesting.h

```diff
@@ -0,0 +1,46 @@
+#include"BMSForCharging.h"
+
+class BatteryMgmtSystmTest : public BatteryManagemtSystem

```

Comment:
The monitoring code needs to be part of production code itself. Is this class pure test code or production code?
The code can be made simpler by removing one level of inheritance. 
My understanding is that this class is production code. Please correct me if it is otherwise.

Reply:
Hi,
> The monitoring code needs to be part of production code itself. Is this class pure test code or production code?
VR: This is pure Test Code.
 >The code can be made simpler by removing one level of inheritance.
VR: Classes BatteryMgmtSystmTest and BatteryMgmtSystmForCharging both inherits from BatteryManagemtSystem (that inherits from Battery). So can you please tell which can be removed.
>My understanding is that this class is production code. Please correct me if it is otherwise.
VR: This is Test Code.

Reply:
Ok clear now. 
Is the test code not testing BatteryMgmtSystmForCharging class? I couldn't find the test code. Please share the file and line number if it is there.

Since BatteryMgmtSystmForCharging is the production class, this is the one that needs to be tested. Hence BatteryMgmtSystmTest could be the one that tests BatteryMgmtSystmForCharging (i.e. this is one possibility where test class contains test code to verify Charging class). therefore the test class can be removed or adapted to contain test code.

Generally we have test classes/stubs to mimic production code that is not available for integration. for example certain hardware or 3rd party components can be tested mostly with only simulators and the hardware is available only in deployment environment. To overcome this restrictions we have stubs. 
Here I understand that the test class is a kind of stub.

Let me know if things are not clear. Always better to get things clarified :-)

Reply:
Yes, Test code was not testing the BatteryMgmtSystmForCharging.
Thanks I have added now.
Yes I used BatteryMgmtSystmTest as a stub.

Reply:
Also, I have added the monitorTemparatureForWarning(),monitorSocForWarning(), monitorChargeRateForWarning() inside isBatteryOk().